eng/mgmt/automation/generate_data.py

```diff
@@ -389,7 +389,7 @@ def generate(
 
 
 def compile_package(sdk_root: str, group_id: str, module: str) -> bool:
-    command = 'mvn --no-transfer-progress clean verify package -f {0}/pom.xml -Dmaven.javadoc.skip -Dgpg.skip -DskipTestCompile -Drevapi.skip -pl {1}:{2} -am'.format(
+    command = 'mvn --no-transfer-progress clean verify package -f {0}/pom.xml -Dmaven.javadoc.skip -Dgpg.skip -Dmaven.test.skip -Drevapi.skip -pl {1}:{2} -am'.format(
```

Comment:
I remember this one in script is recommended by @dw511214992 , to skip compile on test codes (instead of compile but not run), for the case that updated Cadl/Swagger breaks the previous tests (maybe caused by breaking changes).

Reply:
Yes. The compiling failure caused by test codes shouldn't block generating the package and apiview. If there is a way to compile the test and not block generating apiview, it will be better.

Reply:
@alzimmermsft So I guess eng team might want this line remain same.

Alternatively we can run the package twice, one for apiview, one not (which includes tests). But seems not worth the additional complixity.

Reply:
Thanks for pointing this out, I'm going to revert most this PR as only specific executions needs to use the proper skipping parameters.